torch/fx/experimental/validator.py

```diff
@@ -277,12 +277,42 @@ def wrapper(*args):
 
             return wrapper
 
+        # Like `lift`, but for bitwise ops. We need to convert to/from BitVec
+        # in order to use z3 bitwise ops. We assume that integers are 64 bit.
+        def bitwise_lift(func):
+            def wrap(a) -> z3.ExprRef:
+                if isinstance(a, bool):
+                    a = z3.BoolVal(a)
+                if isinstance(a, (int, sympy.Integer)):
+                    a = z3.IntVal(int(a))
+                if not isinstance(a, (z3.ArithRef, z3.BoolRef)):
+                    raise ValueError(f"can't lift type for bitwise op: {type(a)}")
+                try:
+                    return z3.Int2BV(a, 64)
+                except z3.Z3Exception:
+                    # Catch z3 Int/Bool sort error
+                    return z3.Int2BV(a != 0, 64)
+
+            @functools.wraps(func)
+            def wrapper(*args):
+                # Lifts the arguments into a list of Z3 inhabitants.
+                if len(args) == 1 and isinstance(args[0], (list, tuple)):
+                    wrapped_args = (tuple(wrap(a) for a in args[0]),)
+                else:
+                    wrapped_args = tuple(wrap(a) for a in args)
+                # Run the function on the Z3 expressions.
+                return z3.BV2Int(func(*wrapped_args))
+
+            return wrapper
```

Comment:
What do you think about reusing the code in the `lift` function? Both `wrap` and `wrapper` look pretty similar to me. Then, you could just wrap them with a bitwise wrapper.

Reply:
Yep, the 2 functions became more similar after some changes I made.